src/lib.rs

```diff
@@ -1 +1,208 @@
+                    directory,
+                    &startup_info,
+                    &mut process_info as *mut PROCESS_INFORMATION,
+                )
+            } else {
+                windows::Win32::System::Threading::CreateProcessW(
+                    PWSTR::default(),
+                    command,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    inherit_handles,
+                    process_creation_flags,
+                    std::ptr::null() as *const c_void,
+                    PWSTR::default(),
+                    &startup_info,
+                    &mut process_info as *mut PROCESS_INFORMATION,
+                )
+            };
+
+            if res.as_bool() {
+                Ok(Self {
+                    process_information: process_info,
+                })
+            } else {
+                Err(Error::CreationFailed(get_last_error()))
+            }
+        }
+    }
+
+    pub fn wait(&self) -> Result<ExitStatus> {
+        let mut exit_code: u32 = 0;
+
+        unsafe {
+            let res = WaitForSingleObject(self.process_information.hProcess, INFINITE);
+
+            if res == WAIT_OBJECT_0 {
+                if GetExitCodeProcess(
+                    self.process_information.hProcess,
+                    &mut exit_code as *mut u32,
+                )
+                .as_bool()
+                {
+                    close_handle(self.process_information);
+                    Ok(ExitStatus(exit_code))
+                } else {
+                    Err(Error::GetExitCodeFailed(get_last_error()))
+                }
+            } else {
+                Err(Error::GetExitCodeFailed(get_last_error()))
+            }
+        }
+    }
+
+    pub fn try_wait(&self) -> Result<Option<ExitStatus>> {
+        let mut exit_code: u32 = 0;
+
+        let res = unsafe {
+            GetExitCodeProcess(
+                self.process_information.hProcess,
+                &mut exit_code as *mut u32,
+            )
+        };
+
+        if res.as_bool() {
+            if exit_code == STATUS_PENDING.0 {
+                Ok(None)
+            } else {
+                close_handle(self.process_information);
+                Ok(Some(ExitStatus(exit_code)))
+            }
+        } else {
+            Err(Error::GetExitCodeFailed(get_last_error()))
+        }
+    }
+
+    pub fn kill(&self) -> Result<()> {
+        let res = unsafe { TerminateProcess(self.process_information.hProcess, 0) };
+
+        if res.as_bool() {
+            Ok(())
+        } else {
+            Err(Error::KillFailed(get_last_error()))
+        }
+    }
+}
+
+pub struct ExitStatus(u32);
+
+impl ExitStatus {
+    pub fn success(&self) -> bool {
+        self.0 == 0
+    }
+
+    pub fn code(&self) -> u32 {
+        self.0
+    }
+}
+
+fn get_last_error() -> u32 {
+    unsafe { GetLastError().0 }
+}
```

Comment:
Simply wrapping an unsafe function into a safe function doesn't make it magically safe.

Also: pretty useless wrapper!

```suggestion
unsafe fn get_last_error() -> u32 {
    GetLastError().0
}
```